src/python/bin/aqueduct

```diff
@@ -57,6 +57,77 @@ def update_executable_permissions():
     execute_command(["chmod", "755", os.path.join(server_directory, "bin", "executor")])
     execute_command(["chmod", "755", os.path.join(server_directory, "bin", "migrator")])
 
+def generate_version_file(file_path):
+    with open(file_path, 'w') as f:
+        f.write(package_version)
+
+# Returns a bool indicating whether we need to perform a version upgrade.
+def require_update(file_path):
+    if not os.path.isfile(file_path):
+        return True
+    with open(file_path, 'r') as f:
+        current_version = f.read()
+        if package_version < current_version:
+            raise Exception("Attempting to install an older version %s but found existing newer version %s" % (package_version, current_version))
```

Comment:
Are we not allowing users to downgrade our packages at all? What if they want to go back to and older version because of breaking changes we introduced?

Reply:
The problem with downgrading right now is that we haven't implemented the down migration of sqlite. The reason was that in SQLite, lots of column drop and constraint drop operations aren't supported, which makes it pretty tedious to implement down migration; we basically need to create a new table with old constraints and dump data from the current table into that. I think we should eventually still implement this and support it, but not sth we can support right now, so that's why I'm disabling downgrading now.

Reply:
Yeah, that makes sense for why we can't support downgrades right now (without wiping the database.)